pymc3/step_methods/metropolis.py

```diff
@@ -78,6 +81,53 @@ def __call__(self, num_draws=None):
             return np.dot(self.chol, b)
 
 
+class RecursiveDAProposal(Proposal):
+    """
+    Recursive Delayed Acceptance proposal to be used with MLDA step sampler.
+    Recursively calls an MLDA sampler if level > 0 and calls Metropolis
+    sampler if level = 0. Results in a hierarchy of chains each of which is
+    used to propose samples to the chain above.
+    """
+    def __init__(self, next_step_method, next_model,
+                 tune, tune_interval, subsampling_rate):
+
+        self.next_step_method = next_step_method
+        self.next_model = next_model
+        self.tune = tune
+        self.tune_interval = tune_interval
+        self.subsampling_rate = subsampling_rate
+
+    def __call__(self, q0_dict):
+        """Returns proposed sample
+        given the current sample
+        in dictionary form (q0_dict).
+        """
+
+        _log = logging.getLogger('pymc3')
+        _log.setLevel(logging.ERROR)
```

Comment:
This is not for debugging. Because of the recursion, many levels of samplers are called and this prevents them from printing their output, progress bars. Without it the console fills up with unnecessary information. I looked at the pymc3 forums and this is the only way I found to silence the whole output although I suspect there might be smoother ways.

Reply:
👍